### PR TITLE
Fix button flicker.

### PR DESCRIPTION
**Pull request by themithy**

### CODE REVIEW

File: packages/neptune-extension/src/common/components/App.tsx [hunk 1]
```diff
@@ -37,51 +38,71 @@ const App: React.FC<AppProps> = ({
...
 
+  /*
+   * This should be probably considered in detail and we should have one 
+   * initialization flow and one flag guarding that flow.
+   *
+   * 1. no api token - initialization finished, configure button only.
+   * 2. api token invalid - same as 1. but requires async check.
+   * 3. api token valid & notebook loaded
+   */
+  const applicationInitialized = apiToken === undefined || isApiTokenValid === false || (isApiTokenValid === true && notebookInitialized);
```
Comment (kierson) [inline packages/neptune-extension/src/common/components/App.tsx]: I think this is complicated.

I propose simpler solution.

1. Remove `notebookInitialized` completely from App and move its logic to upload modal (show local loader indicator if needed).
2. `const applicationInitialized` = apiToken === undefined || isApiTokenValid !== undefined.

`isApiTokenValid` is undefined at the begining.
So:
a) if we dont have API Token, app is initialized.
b) if we have `apiToken` we should check if `isApiTokenValid` is undefined. If not, then app is also initialized.

This is much simpler.
Plus we don't need loading indicator for whole app, we just need one in upload modal.

I would not like to see global loading indicator triggered by one of my installed extensions.

Comment (themithy) [reply packages/neptune-extension/src/common/components/App.tsx]: The whole application depends on the notebook check (aka loadInitialNotebook).
1. Upload - decide upload notebook/checkpoint scenario.
2. Checkout - populate project id and notebook id.
3. Automatic activation - set envs in the bg when notebook is open.
4. Manual activation - display the activation button or not.

Comment (themithy) [reply packages/neptune-extension/src/common/components/App.tsx]: One way or the other, so far this changes only defer button loading, until the state of application is known. This prevents button flicker and enforces the delay of other actions.

Whether we show spinner of not, it's different story and not a part of this PR - see The `AppLoader` is empty right now.

Comment (kierson) [reply packages/neptune-extension/src/common/components/App.tsx]: If it is empty, then remove it from this PR.

I still think that `applicationInitialized` has too complicated condition.
`const applicationInitialized = apiToken === undefined || (notebookInitialized &&  isApiTokenValid !== undefined)`

This is simpler.

**Pull request merged**